pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -0,0 +1,859 @@
+                    signed=False
+                ) - waveform_preamble.horizontal_offset
+
+                coord_x = max(coord_x, 0)
+
+                coord_y = int.from_bytes(self.raw_points_coordinates
+                                         [i + self.n_bytes_for_x_coordinate:
+                                          i + self.n_bytes_for_x_coordinate +
+                                          self.n_bytes_for_y_coordinate],
+                                         byteorder="big",
+                                         signed=False
+                                         ) - waveform_preamble.vertical_offset
+                coord_y = max(coord_y, 0)
+
+                self.points_coordinates.append((coord_x, coord_y))
+                self.points.append((coord_x * waveform_preamble.x_scale_factor,
+                                    coord_y * self.waveform_preamble.y_scale_factor))
+
+    def __str__(self):
+
+        description = f'WAVEFORM DATA \n' \
+                      f'{self.waveform_preamble}' \
+                      f'The Waveform Curve Data is:\n' \
+                      f'Curve Coordinate Points: {self.points_coordinates} \n' \
+                      f'Curve Points: {self.points} \n\n'
+
+        return description
+
+
+class Tektronix371A(Instrument):
+    """ Represents the Tektronix Curve Tracer model 371A
+    and provides a high-level interface for interacting with
+    the instrument using the SCPI command set.
+
+    .. code-block:: python
+
+    ct371A = Tektronix371A("GPIB0::23::INSTR")
+    # COLLECTOR SUPPLY
+    ct371A.cs_peakpower = 300
+    ct371A.cs_polarity = "POS"
+    ct371A.cs_collector_supply = 0
+    # STEP GEN
+    ct371A.stepgen_step_source_and_size = ("VOLTAGE", 5.0)
+    ct371A.stepgen_number_steps = 0
+    ct371A.set_stepgen_offset(0)
+    # DISPLAY
+    ct371A.diplay_store_mode = "STO"
+    ct371A.display_horizontal_source_sensitivity = ("COLLECT", 1.0E-1)
+    ct371A.display_vertical_source_sensitivity = ("COLLECT", 500.0E-3)
+    ct371A.set_cursor_mode("DOT", 1)
+    # MEASUREMENT
+    ct371A.measure_mode = "REP"
+
+    """
+
+    COLLECTOR_SUPPLY_POLARITY_MODES = ["NPN", "POSitive", "PNP", "NEGative", "POS", "NEG"]
+    COLLECTOR_SUPPLY_PEAKPOWER_VALUES = [3000, 300, 30, 3]  # in watts
+    COLLECTOR_SUPLLY_RANGE = [0.0, 100.0]  # in % of max peak power
+    CURSOR_DOT_SET = list(range(0, 1025, 1))  # 0 is the beginning and 1024 is the end of the curve
+    HORIZONTAL_SOURCE_SET = ["STPgen", "STP", "COLlect", "COL"]
+    VERTICAL_SOURCE_SET = ["COLlect", "COL"]
+    STEP_GENERATOR_SOURCE_SET = ["VOLtage", "CURrent", "VOL", "CUR"]
+    DISPLAY_VALID_STORE_MODES = ["NST", "STO", "NSTore", "STOre"]
+    VALID_MEASUREMENT_MODES = ["REPeat", "REP", "SINgle", "SIN", "SWEep", "SWE", "SSWeep", "SSW"]
+    VALID_CURSOR_MODES = ["OFF", "DOT", "LINE", "WINDOW"]
+    VALID_V_H_COORDINATES_SET = list(range(0, 1025, 1))  # 0 is the beginning
+    # and 1023 is the end of the valid coordinates set
```

Comment:
if 1023 is the last valid value, you should use `range(1024)`, which is equal to `range(0, 1024, 1)` as the first 0 and the last 1 are default values.
```suggestion
    VALID_V_H_COORDINATES_SET = list(range(1024))  # 0 is the beginning
    # and 1023 is the end of the valid coordinates set
```

Reply:
You are right. In fact the A/D is 10 bits, so the scale is 0 - 1023. Many thanks